src/python/pants/backend/tools/trufflehog/rules.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2024 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+""""Generates trufflehog rules."""
+
+from __future__ import annotations
+
+from pants.core.goals.lint import LintFilesRequest, LintResult
+from pants.core.util_rules.config_files import ConfigFiles, ConfigFilesRequest
+from pants.core.util_rules.external_tool import DownloadedExternalTool, ExternalToolRequest
+from pants.core.util_rules.partitions import Partitions
+from pants.engine.fs import (
+    CreateDigest,
+    Digest,
+    DigestEntries,
+    FileEntry,
+    MergeDigests,
+    PathGlobs,
+    Snapshot,
+)
+from pants.engine.platform import Platform
+from pants.engine.process import FallibleProcessResult, Process
+from pants.engine.rules import Get, MultiGet, collect_rules, rule
+from pants.source.filespec import FilespecMatcher
+from pants.util.logging import LogLevel
+from pants.util.strutil import pluralize
+
+from .subsystem import Trufflehog
+
+
+class TrufflehogRequest(LintFilesRequest):
+    tool_subsystem = Trufflehog
+
+
+@rule
+async def partition_inputs(
+    request: TrufflehogRequest.PartitionRequest, trufflehog: Trufflehog
+) -> Partitions:
+    """Configure the partitions scheme."""
+    if trufflehog.skip or not request.files:
+        return Partitions()
+
+    matched_filepaths = FilespecMatcher(
+        includes=["**"],
+        excludes=trufflehog.exclude_options,
+    ).matches(tuple(request.files))
+    return Partitions.single_partition(sorted(matched_filepaths))
```

Comment:
Is trufflehog fast? Is there any benefit to partitioning in a more fine-grained way to get more concurrency and finer-grained caching? With a single partition you get only very coarse-grained caching, since if any file in the partition changes the entire partition must re-run. But for many tools that doesn't matter since they're extremely fast even on thousands of files.

Reply:
I would say trufflehog is relatively speedy...... I just ran `trufflehog git file://.` on the entire pants codebase and the scan took 16 seconds........for our very chunky pants repository, the full file scan takes 5 seconds

Reply:
Open to partitioning further if that 16 seconds is outside of an SLO for backend speed

Reply:
I think it's fine for now! We can always fix in a followup if there is demand.

What we typically want to do is run a process per file, because then we get the finest caching granularity. But sometimes that is ridiculous (e.g., ruff, where the actual per-file execution time is deep in the noise compared to Pants process execution overhead). This is why we have batching as an option.